sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -151,6 +175,73 @@
             });
     }
 
+    private Flux<Instant> renewMessageLock(ServiceBusReceivedMessage[] messagesforRenewLock) {
+
+        return  cbsAuthorizationOnce()
+            .then(
+                channelMono.flatMap(requestResponseChannel -> {
+                    UUID[] lockTokens = new UUID[messagesforRenewLock.length];
+                    int index = 0;
+                    for (ServiceBusReceivedMessage receivedMessage : messagesforRenewLock
+                    ) {
+                        lockTokens[index++] = receivedMessage.getLockToken();
+                    }
+                    HashMap<String, UUID[]> requestBodyMap = new HashMap<>();
+                    requestBodyMap.put(REQUEST_RESPONSE_LOCKTOKENS, lockTokens);
+                    Message requestMessage = createRequestMessageFromValueBody(REQUEST_RESPONSE_RENEWLOCK_OPERATION,
+                        requestBodyMap, MessageUtils.adjustServerTimeout(Duration.ofSeconds(60)));
+                    return requestResponseChannel.sendWithAck(requestMessage);
+                })
+            )
+            .flatMapMany(responseMessage -> {
+                   int statusCode = RequestResponseUtils.getResponseStatusCode(responseMessage);
+
+                    List<Instant> expirationsForLocks = null;
+
+                    if (statusCode ==  REQUEST_RESPONSE_OK_STATUS_CODE) {
+                        @SuppressWarnings("unchecked")
+                        Map<String, Object> responseBody = (Map<String, Object>) ((AmqpValue) responseMessage
+                            .getBody()).getValue();
+                        Object expirationListObj = responseBody.get(REQUEST_RESPONSE_EXPIRATIONS);
+
+                        if (expirationListObj instanceof  Date[]){
+                            Date[] expirations = (Date[]) expirationListObj;
+                                expirationsForLocks =  Arrays.stream(expirations)
+                                .map(Date::toInstant)
+                                .collect(Collectors.toList());
+                        }
+                    }
+                    return Flux.fromIterable(expirationsForLocks);
```

Comment:
If `expirationsForLocks` remains `null` this will throw a `NullPointerException`, should guard this and return `Flux.empty` if we get back a bad response.